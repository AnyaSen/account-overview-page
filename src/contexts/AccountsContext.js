import React, { useState, createContext } from "react";

export const AccountsContext = createContext();

export const AccountsContextProvider = ({ children }) => {
  const [accNumber, setAccNumber] = useState(90);
  const [accName, setAccName] = useState("FUNDING_ACCOUNT");
  const [accID, setAccID] = useState("DKUW67");
  const [accBalance, setAccBalance] = useState("60â‚¬");

  return (
    <AccountsContext.Provider
      value={{
        accNumber,
        accName,
        accID,
        accBalance
      }}
    >
      {children}
    </AccountsContext.Provider>
  );
};
