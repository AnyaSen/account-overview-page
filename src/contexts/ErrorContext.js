import React, { createContext, useState } from "react";

export const ErrorContext = createContext();

export const ErrorContextProvider = ({ children }) => {
  const [isAccountsError, setIsAccountsError] = useState(false);
  const [isUserInfoError, setIsUserInfoError] = useState(false);

  return (
    <ErrorContext.Provider
      value={{
        isAccountsError,
        setIsAccountsError,
        isUserInfoError,
        setIsUserInfoError
      }}
    >
      {children}
    </ErrorContext.Provider>
  );
};
