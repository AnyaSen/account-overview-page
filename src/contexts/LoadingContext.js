import React, { createContext, useState } from "react";

export const LoadingContext = createContext();

export const LoadingContextProvider = ({ children }) => {
  const [isAccountsLoading, setIsAccountsLoading] = useState(true);
  const [isUserInfoLoading, setIsUserInfoLoading] = useState(true);

  return (
    <LoadingContext.Provider
      value={{
        isAccountsLoading,
        setIsAccountsLoading,
        isUserInfoLoading,
        setIsUserInfoLoading
      }}
    >
      {children}
    </LoadingContext.Provider>
  );
};
