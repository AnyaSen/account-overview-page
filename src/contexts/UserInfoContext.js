import React, { useState, createContext } from "react";

export const UserInfoContext = createContext();

export const UserInfoContextProvider = ({ children }) => {
  const [name, setName] = useState("SOME NAME");

  return (
    <UserInfoContext.Provider
      value={{
        name
      }}
    >
      {children}
    </UserInfoContext.Provider>
  );
};
