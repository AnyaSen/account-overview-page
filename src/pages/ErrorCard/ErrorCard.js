import React from "react";
import { Link } from "react-router-dom";

import Styles from "./ErrorCard.module.scss";

import errorIcon from "../../assets/img/error.png";
import SecondaryButton from "../../components/Buttons/SecondaryButton/SecondaryButton";

export default function ErrorCard() {
  return (
    <div className={Styles.ErrorCardContainer}>
      <img src={errorIcon} alt="Error" />

      <h3>Sorry, something went wrong.</h3>

      <Link to="/">
        <SecondaryButton btnText="Go back" />
      </Link>
    </div>
  );
}
