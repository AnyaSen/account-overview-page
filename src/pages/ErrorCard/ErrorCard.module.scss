@import "../../assets/scss/variables";
@import "../../assets/scss/breakpoints";

.ErrorCardContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  padding: $space-1;
  text-align: center;

  img {
    height: 9rem;
    margin-bottom: $space-4;

    @include sm {
      height: 7rem;
    }
  }
  h3 {
    margin-bottom: $space-3;
  }
}
