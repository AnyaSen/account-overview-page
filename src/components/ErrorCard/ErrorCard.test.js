import React from "react";
import { render, cleanup } from "@testing-library/react";

import ErrorCard from "./ErrorCard";

import { LoadingContext } from "../../contexts/LoadingContext";
import { ErrorContext } from "../../contexts/ErrorContext";
import { AccountsContext } from "../../contexts/AccountsContext";

const createRenderTree = (isAccountsError, isUserInfoError, errorMessages) => (
  <LoadingContext.Provider value={{ isAccountsLoading: false }}>
    <ErrorContext.Provider
      value={{
        isAccountsError,
        isUserInfoError,
        errorMessages
      }}
    >
      <AccountsContext.Provider value={{ modifiedAccounts: [] }}>
        <ErrorCard />
      </AccountsContext.Provider>
    </ErrorContext.Provider>
  </LoadingContext.Provider>
);

describe("<ErrorCard />", () => {
  afterAll(cleanup);

  const errorMessages = {
    accountsError: "Sorry, could not get the accounts data",
    userInfoError: "Sorry, could not get the user data"
  };

  test("Should contain the correct image and shows error message for accounts", () => {
    const { getByTestId, getByAltText, container } = render(
      createRenderTree(true, false, errorMessages)
    );

    const errorIcon = getByAltText("Error");
    const errortext = getByTestId("errorMessage");

    expect(errorIcon.src).toContain("error.png");
    expect(errortext.textContent).toBe(errorMessages.accountsError);

    expect(container.firstChild).toMatchSnapshot();
  });

  test("Should show error message for userInfo", () => {
    const { getByTestId } = render(
      createRenderTree(false, true, errorMessages)
    );

    const errortext = getByTestId("errorMessage");

    expect(errortext.textContent).toBe(errorMessages.userInfoError);
  });
});
