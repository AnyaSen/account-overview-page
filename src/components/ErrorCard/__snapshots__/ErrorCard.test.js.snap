// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorCard /> Should contain the correct image and shows error message for accounts 1`] = `
<div
  class="ErrorCardContainer"
>
  <img
    alt="Error"
    src="error.png"
  />
  <h3
    data-testid="errorMessage"
  >
    Sorry, could not get the accounts data
  </h3>
</div>
`;
